internal/serverinstall/ecs.go

```diff
@@ -0,0 +1,2177 @@
+			ExecutionRoleArn:        taskDef.ExecutionRoleArn,
+			Family:                  taskDef.Family,
+			Memory:                  taskDef.Memory,
+			NetworkMode:             aws.String("awsvpc"),
+			RequiresCompatibilities: []*string{aws.String(defaultTaskRuntime)},
+			Tags:                    taskTags,
+			Volumes:                 taskDef.Volumes,
+		}
+
+		ecsSvc := ecs.New(sess)
+		taskDef, err := registerTaskDefinition(&registerTaskDefinitionInput, ecsSvc)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = ecsSvc.UpdateService(&ecs.UpdateServiceInput{
+			Cluster:        &clusterArn,
+			TaskDefinition: taskDef.TaskDefinitionArn,
+			Service:        serverSvc.ServiceName,
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = ecsSvc.WaitUntilServicesStable(&ecs.DescribeServicesInput{
+			Cluster:  &clusterArn,
+			Services: []*string{serverSvc.ServiceName},
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var contextConfig clicontext.Config
+	var advertiseAddr pb.ServerConfig_AdvertiseAddr
+	advertiseAddr.Addr = serverCfg.Address
+	advertiseAddr.Tls = true
+	advertiseAddr.TlsSkipVerify = true
+	contextConfig = clicontext.Config{
+		Server: serverCfg,
+	}
+	httpAddr := strings.Replace(serverCfg.Address, "9701", "9702", 1)
+
+	s.Done()
+	return &InstallResults{
+		Context:       &contextConfig,
+		AdvertiseAddr: &advertiseAddr,
+		HTTPAddr:      httpAddr,
+	}, nil
+}
+
+// Uninstall is a method of ECSInstaller and implements the Installer interface
+// to remove a waypoint-server statefulset and the associated PVC and service
+// from a ecs cluster
+func (i *ECSInstaller) Uninstall(
+	ctx context.Context,
+	opts *InstallOpts,
+) error {
+	ui := opts.UI
+	log := opts.Log
+
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Uninstalling Server resources...")
+	defer func() { s.Abort() }()
+
+	// Get list of resources created with either the waypoint-server, or
+	// waypoint-runner tag
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: i.config.Region,
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+	rgSvc := resourcegroups.New(sess)
+
+	query := fmt.Sprintf(serverResourceQuery, defaultServerTagName)
+	results, err := rgSvc.SearchResources(&resourcegroups.SearchResourcesInput{
+		ResourceQuery: &resourcegroups.ResourceQuery{
+			Type:  aws.String(resourcegroups.QueryTypeTagFilters10),
+			Query: aws.String(query),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	resources := results.ResourceIdentifiers
+
+	// Start destroying things. Some cannot be destroyed before others. The
+	// general order to destroy things:
+	// - ECS Service
+	// - ECS Cluster
+	// - Cloudwatch Log Group
+	// - ELB Target Groups
+	// - ELB Network Load Balancer
+	// - EFS File System
+
+	s.Update("Deleting ECS resources...")
+	if err := deleteEcsResources(ctx, sess, resources); err != nil {
+		return err
+	}
+	s.Done()
+
+	s = sg.Add("Deleting Cloud Watch Log Group resources...")
```

Comment:
Maybe instead of a new stepgroup we can just make this an `Update` so it just is all 1 line of output when these resources get deleted? If not, maybe add an `Update` after the deletion to change `Deleting` to `Deleted` 🤔

Reply:
Agreed, done in 6b212967